packages/hooks/src/use_shared_state.rs

```diff
@@ -121,30 +121,98 @@ pub struct UseSharedState<T> {
     pub(crate) inner: Rc<RefCell<ProvidedStateInner<T>>>,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum UseSharedStateError {
```

Comment:
It would be nice to provide information about the location where the state was previously read and written to in debug mode to provide nicer debug information. (This would use https://doc.rust-lang.org/reference/attributes/codegen.html#the-track_caller-attribute to track the file location of the caller of read and write in debug mode)

(It doesn't need to implemented in this PR)

Reply:
oooh I've never seen that one - nice. (pushed)

Reply:
pushed another one, so that it's only there in `debug` mode

Reply:
oh yeah, tracking itself should also be debug-only - pushed

Reply:
Apparently what I was thinking of is already implemented in https://docs.rs/debug-cell/latest/debug_cell. It tracks the current borrows and logs the location of any relevant borrows on errors

Reply:
hmmm this could almost be a separate feature behind a specific feature flag, idk your call

Reply:
It is probably better implemented in a different PR, but I think we could enable it in debug mode by default

Reply:
want me to do it?

Reply:
If you are up for it, that would be great! We could also use the same method for the use_ref hook

Reply:
ok soo I've done something a bit different instead but with the same result... let me know what you think

Reply:
soo the problem is... it doesn't work

```rust
Writing to shared state failed: [/home/niedzwiedz/.cargo/git/checkouts/dioxus-7f2b82817ca79561/760d937/packages/hooks/src/use_shared_state.rs:296:27] dioxus_hooks::use_shared_state::UseSharedState<editor::app::App> is already borrowed at [None], so it cannot be borrowed mutably.
```

as you can see it shows the spot inside `hooks/src/use_shared_state.rs` ...

Reply:
ok it doesn't actually work, it only shows the thing that was previously borrowed but it could've been dropped in the meantime. I guess there's no way around it other than rolling a custom RefCell that tracks borrows but also when they are released

Reply:
https://github.com/alexcrichton/debug-cell/pull/5 made a PR to `debug-cell` to have it expose the `try_*` methods on debug RefCell, let's hope it's not too much to ask 🤞🏿